ExScripts-NickTracker/ExtioNickTrackerVer5-6.kvs

```diff
@@ -0,0 +1,599 @@
+echo Installing NickTracker By Extio v5.6
+
+echo Removing any old aliases, aliases now namespaced (only if previously installed).
+alias -q (NTCloneScanProcessAll) {};
+alias -q (NTCloneScanProcess) {};
+alias -q (NTCloneScan) {};
+alias -q (NickTrackerProcess) {};
+
+echo building CloneScan Processes;
+alias(NT::CloneScanProcess)
+{
+	%chanUsersArray[] = $chan.users($channel($chan.name),,m) ;
+	%r = 0; %s = 0;
+	%clonetext="";
+	while (%r < %chanUsersArray[]#)
+	{
+		%chanUsersArrayPieces[] = $str.split([!@],%chanUsersArray[%r],r) ;
+		if ( $1 == %chanUsersArrayPieces[2] && $0 != %chanUsersArrayPieces[0]) 
+		{
+			if (%s == 0)
+			{
+				%clonetext .="\Clones: \[$0\] and \[%chanUsersArrayPieces[0]\]";
+				%s++;
+			} else {
+				%clonetext .= " and \[%chanUsersArrayPieces[0]\]";
+				%s++;
+			};
+		};
+		%r++;
+	};
+	if (%s>0) { echo -i=$msgtype(genericwarning) %clonetext" \>\> \[$1\]"; };
+}
+
+
+alias(NT::CloneScan)
+{
+	%chanUsersArray[] = $chan.users($channel($chan.name),,m) ;
+	%j = 0 ;
+	# echo chanUsersArray: %chanUsersArray[] ;
+	while (%j < %chanUsersArray[]#)
+	{
+		%clonetext="";
+		%chanUsersArrayPieces[] = $str.split([!@],%chanUsersArray[%j],r) ;
+		%g = $(%j+1);
+		%s = 0;
+		while (%g < %chanUsersArray[]#)
+		{
+			if(%chanUsersArray[%j] != ".") # this checks if the nick has been deleted, because it already matched.
+			{
+				%chanUsersArrayPieces2[] = $str.split([!@],%chanUsersArray[%g],r) ;
+				if ( %chanUsersArrayPieces[2] == %chanUsersArrayPieces2[2] ) 
+				{
+					if (%s == 0)
+					{
+						%clonetext .="\Clones: \[%chanUsersArrayPieces[0]\] and \[%chanUsersArrayPieces2[0]\]";
+						%s++;
+					} else {
+						%clonetext .= " and \[%chanUsersArrayPieces2[0]\]";
+						%s++;
+					};
+					%chanUsersArray[%g] = "."; # this deletes tokens from chanusersarray so that when the second loop comes to them, they wont be checked, because a match has already been found.
+				};
+			};
+			%g++;
+		};
+		%j++;
+		if (%s>0) { echo -i=$msgtype(genericwarning) %clonetext" \>\> \[%chanUsersArrayPieces[2]\]"; };
+	};
+}
+
+
+echo Building Nicktracker Process;
+alias(NT::NickTrackerProcess)
+{
+	%NTSet = $config.open(NickTrackerSettings.kvs) ;
+	%nOfN = $config.read(%NTSet,NickNum,6) ;
+	%ntUserDisplay = $config.read(%NTSet,NTdisplayUser,"on");
+	%ntIpDisplay = $config.read(%NTSet,NTdisplayIp,"on");
+	%powerstate = $config.read(%NTSet,NTpower,"on")
+	config.close %NTSet ; 
+	config.setsection %NT $my.network
+	config.setsection %NT2 $my.network
+	%nicksread = $config.read(%NT,$2,"") ;
+	%nicksreadUserName = $config.read(%NT2,$1,"") ;
+	%nicks = "";
+	%nicksUserName = "";
+	if (%nicksread != "") {
+		%nicksarray[] = $str.split(" ",%nicksread,n) ;
+		# echo full array: %nicksarray[];
+		%i = 0 ;
+		while (%i < %nicksarray[]#)
+		{
+			if (%nicksarray[%i] != $0 && %i < $(%nOfN+1))
+			{
+				%nicks << %nicksarray[%i] ;
+			} ;
+			%i++ ;
+		} ;
```

Comment:
You arent though, theres a gazillion semi colons after opening brackets in this

let me show you 3 examples out of many more